packages/examples/cvat/exchange-oracle/tests/api/test_exchange_api.py

```diff
@@ -1,100 +1,306 @@
+        open("tests/utils/manifest.json") as data,
+        patch("src.endpoints.serializers.get_escrow_manifest") as mock_get_manifest,
+    ):
+        manifest = json.load(data)
+        mock_get_manifest.return_value = manifest
+
+        required_fields = {
+            "escrow_address",
+            "chain_id",
+            "job_type",
+            "status",
+            "qualifications",
+            "updated_at",
+        }
+        selectable_fields = {"job_description", "created_at", "reward_amount", "reward_token"}
+
+        all_schema_fields = required_fields | selectable_fields
+
+        for fields in [None] + [
+            list(combination)
+            for fields_len in range(1, len(selectable_fields) + 1)
+            for combination in combinations(selectable_fields, fields_len)
+        ]:
             response = client.get(
-                "/tasks", headers={"signature": "sample"}, params={"wallet_address": user_address}
+                "/job",
+                headers=get_auth_header(),
+                params={"fields": ",".join(fields)} if fields else None,
             )
-
             assert response.status_code == 200
-            assert isinstance(response.json(), list)
-            assert len(response.json()) == 1
-            for task in response.json():
-                assert task["assignment"]
-                assert set(task.keys()) == {
-                    "id",
-                    "escrow_address",
-                    "title",
-                    "description",
-                    "platform",
-                    "job_bounty",
-                    "job_size",
-                    "job_time_limit",
-                    "job_type",
-                    "assignment",
-                    "status",
-                }
+            jobs = response.json()["results"]
+            assert set(jobs[0].keys()) == (
+                (required_fields | set(fields)) if fields else all_schema_fields
+            )
+
+    session.close()
 
 
-def test_list_tasks_200_without_address(client: TestClient) -> None:
+def test_list_jobs_200_check_values(client: TestClient) -> None:
+    session = SessionLocal()
+    session.begin()
+    user = User(
+        wallet_address=user_address,
+        cvat_email=cvat_email,
+        cvat_id=1,
+    )
+    session.add(user)
+
+    cvat_project, cvat_task, cvat_first_job = create_project_task_and_job(
+        session, "0x86e83d346041E8806e352681f3F14549C0d2BC66", 1
+    )
+
+    cvat_second_job = create_job(session, 2, cvat_task.cvat_id, cvat_project.cvat_id)
+    session.commit()
+
+    for job in (cvat_second_job, cvat_first_job):
+        assignment = Assignment(
+            id=str(uuid.uuid4()),
+            user_wallet_address=user_address,
+            cvat_job_id=job.cvat_id,
+            expires_at=datetime.now() + timedelta(days=1),
+        )
+        session.add(assignment)
+        job.status = JobStatuses.in_progress
+        session.commit()
+        sleep(0.5)
```

Comment:
Why this sleep is needed?

Reply:
This test should also check that `updated_at` will correspond to the latest updated job. Therefore I emulate sequential job updates by calling `session.commit()` after each change. Using `sleep` is optional, deleted that.